tests/jballerina-unit-test/src/test/resources/test-src/expressions/mappingconstructor/spread_op_field.bal

```diff
@@ -246,6 +246,50 @@ function testSpreadFieldWithRecordTypeHavingNeverField() {
     assertEquality(8, bar8.i);
 }
 
+type Vehicle record {|
+    int year;
+    string manufacturer;
+    string model;
+    anydata...;
+|};
+
+type Truck record {
+    int year;
+    string manufacturer;
+    string model;
+    int loadCapacity;
+};
+
+type RecA record {|
+    int i;
+    string s;
+    string m;
+    any|error...;
+|};
+
+type RecB record {
+    int i;
+    string s;
+    string m;
+    error e;
+};
+
+function testSpreadFieldWithRecordTypeHavingRestDescriptor() {
+    Truck truck = {year: 2023, manufacturer: "Tesla", model: "Cybertruck", loadCapacity: 15000};
+    Vehicle vehicle = {...truck};
+    assertEquality("Cybertruck", vehicle.model);
+    assertEquality("Tesla", vehicle.manufacturer);
+    assertEquality(2023, vehicle.year);
+    assertEquality(15000, vehicle["loadCapacity"]);
+
+    RecB recB = {i: 1, s: "s", m: "m", e: error("e")};
```

Comment:
Shall we also add a test where the source and target are both open records with explicit rest fields?